backend/data_adapter.go

```diff
@@ -45,13 +45,19 @@ func (a *dataSDKAdapter) QueryData(ctx context.Context, req *pluginv2.QueryDataR
 				continue
 			}
 
-			// if error source not set and the error is a downstream error, set error source to downstream.
-			if !r.ErrorSource.IsValid() && IsDownstreamError(r.Error) {
-				r.ErrorSource = ErrorSourceDownstream
+			if !r.ErrorSource.IsValid() {
+				// if the error is a downstream error, set error source to downstream, otherwise plugin.
+				if IsDownstreamError(r.Error) {
+					r.ErrorSource = ErrorSourceDownstream
+				} else {
+					r.ErrorSource = ErrorSourcePlugin
+				}
+				resp.Responses[refID] = r
```

Comment:
Found out that we have to re-assign the response to the map for the updated `ErrorSource` to "stick". Seems like a pretty important bug fix FYI @ivanahuckova

Reply:
This seems odd - would that not have been super obvious before? 🤔 Wonder is it nicer (and valid) to do instead
```go
resp.Responses[refID].ErrorSource = ErrorSourceDownstream
```

Reply:
Ah maybe because it will still log the correct one but just not pass it through! 🤦

Reply:
Yeah exactly 🌟

Reply:
> ```go
> resp.Responses[refID].ErrorSource = ErrorSourceDownstream
> ```

Apparently you're not allowed to do that "cannot assign to struct field resp.Responses[refID].ErrorSource in mapcompiler[UnaddressableFieldAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#UnaddressableFieldAssign)"